infrastructure/terraform/functions/main.tf

```diff
@@ -11,52 +11,58 @@ resource "azurerm_storage_account" "fxnstor" {
   account_kind             = "StorageV2"
 }
 
-resource "azurerm_app_service_plan" "fxnapp" {
+resource "azurerm_service_plan" "fxnapp" {
   name                = format("%s-%s-fxn-asp-%s", var.appname, var.domainprefix, var.environment)
   location            = var.location
   resource_group_name = var.resource_group_name
-  kind                = "functionapp"
-  sku {
-    tier = "Dynamic"
-    size = "Y1"
-  }
+  os_type             = "Windows"
+  sku_name            = "Y1"
 }
 
-resource "azurerm_function_app" "fxn" {
+resource "azurerm_windows_function_app" "fxn" {
   name                       = format("%s-%s-fxn-%s", var.appname, var.domainprefix, var.environment)
   location                   = var.location
   resource_group_name        = var.resource_group_name
-  app_service_plan_id        = azurerm_app_service_plan.fxnapp.id
+  service_plan_id            = azurerm_service_plan.fxnapp.id
   storage_account_name       = azurerm_storage_account.fxnstor.name
   storage_account_access_key = azurerm_storage_account.fxnstor.primary_access_key
-  version                    = "~3"
-  https_only                 = true
-
+  functions_extension_version = "~4"
+  https_only                  = true
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME = "dotnet"
+  }
+  site_config {
+  }
   identity {
     type = "SystemAssigned"
   }
-
   lifecycle {
     ignore_changes = [
-      app_settings
+      app_settings,
+      site_config
     ]
   }
 }
 
-resource "azurerm_function_app_slot" "fxnslot" {
+resource "azurerm_windows_function_app_slot" "fxnslot" {
```

Comment:
Why the name "azurerm_windows_function_app" instead of "azurerm_function_app" ? Is it running in Windows container?

Reply:
Because it is deprecated. See https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/function_app
Yes, I checked on a previous deployment (dubcard), it runs on Windows.